jrnl/jrnl.py

```diff
@@ -320,6 +338,18 @@ def _delete_search_results(journal, old_entries, **kwargs):
         journal.write()
 
 
+def _change_time_search_results(args, journal, old_entries, **kwargs):
+    # separate entries we are not editing
```

Comment:
Can we add something similar here to the start of `_delete_search_results`? That way, if there are no search results, the user will get a message about how nothing is happening. Right now, this doesn't do anything, and is quiet about it, which can be confusing.

Reply:
I wondered about that; I wasn't sure if `--delete` was only warned because it was destructive (for example, `--edit` doesn't warn if nothing is selected), and I reasoned it wasn't exactly an error situation (we're technically performing successfully what was requested).

But I agree it has the potential to be confusing. I'll add it in!

Reply:
I added the message. I used the language "No entries to modify" because it sounded cleaner than "No entries to change time for" or similar, and it has the potential to be used in other situations. But I'm happy to use whatever you suggest.

Unfortunately, I realized that if you run `jrnl --change-time` without anything else, it changes the time of **every entry**. So I'll fix that. In the meantime, I added a test for it (that currently fails).

Reply:
After considering this bug, I posted on #1429 with some considerations about how `--change-time` might benefit from some safeguards.

Reply:
Per @micahellison in #1429, I added a confirmation dialog to `--change-time`. There is one exception: if `--edit` and `--change-time` are used together with only one entry, no confirmation is requested. To my mind, this makes the amend-style action of `--change-time` smooth, while retaining the safety for larger operations.

But I can also see a clear case being made for confirmation even in that case (consistency, confirming what is _still_ a destructive action, albeit on one entry, etc.), so I'm happy to remove that exception. I'm also wondering about adding some sort of output, e.g. "Changing time from X to Y", so the user can at least manually undo an error.